Spurious "Identifier 'angular' must be imported from a module" error

**TypeScript Version:**  

2.0.2

**Code**

https://github.com/jonrimmer/typescript-bug

```
npm install
./node_modules/.bin/tsc bug.ts
```

**Expected behavior:**

It should compile without problems.

**Actual behavior:** 

It gives the following error:

```
node_modules/angular-ui-router/commonjs/ng1/services.d.ts(2,27): error TS2686: Identifier 'angular' must be imported from a module
```

This is despite the fact that the @type/angular definitions includes a global angular var declaration.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -914,8 +914,8 @@ namespace ts {
                     }
                 }
 
-                // If we're in an external module, we can't reference symbols created from UMD export declarations
-                if (result && isInExternalModule) {
+                // If we're in an external module, we can't reference value symbols created from UMD export declarations
+                if (result && isInExternalModule && (meaning & SymbolFlags.Value) === SymbolFlags.Value) {
                     const decls = result.declarations;
                     if (decls && decls.length === 1 && decls[0].kind === SyntaxKind.NamespaceExportDeclaration) {
                         error(errorLocation, Diagnostics.Identifier_0_must_be_imported_from_a_module, name);

--- a/tests/baselines/reference/umd8.errors.txt
+++ b/tests/baselines/reference/umd8.errors.txt
@@ -1,0 +1,25 @@
+tests/cases/conformance/externalModules/a.ts(7,14): error TS2686: Identifier 'Foo' must be imported from a module
+
+
+==== tests/cases/conformance/externalModules/a.ts (1 errors) ====
+    /// <reference path="foo.d.ts" />
+    import * as ff from './foo';
+    
+    let y: Foo; // OK in type position
+    y.foo();
+    let z: Foo.SubThing; // OK in ns position
+    let x: any = Foo; // Not OK in value position
+                 ~~~
+!!! error TS2686: Identifier 'Foo' must be imported from a module
+    
+==== tests/cases/conformance/externalModules/foo.d.ts (0 errors) ====
+    
+    declare class Thing {
+    	foo(): number;
+    }
+    declare namespace Thing {
+    	interface SubThing { }
+    }
+    export = Thing;
+    export as namespace Foo;
+    

--- a/tests/baselines/reference/umd8.js
+++ b/tests/baselines/reference/umd8.js
@@ -5,17 +5,25 @@
 declare class Thing {
 	foo(): number;
 }
+declare namespace Thing {
+	interface SubThing { }
+}
 export = Thing;
 export as namespace Foo;
 
 //// [a.ts]
 /// <reference path="foo.d.ts" />
-let y: Foo;
-y.foo();
+import * as ff from './foo';
 
+let y: Foo; // OK in type position
+y.foo();
+let z: Foo.SubThing; // OK in ns position
+let x: any = Foo; // Not OK in value position
 
 
 //// [a.js]
-/// <reference path="foo.d.ts" />
-var y;
+"use strict";
+var y; // OK in type position
 y.foo();
+var z; // OK in ns position
+var x = Foo; // Not OK in value position

--- a/tests/cases/conformance/externalModules/umd8.ts
+++ b/tests/cases/conformance/externalModules/umd8.ts
@@ -5,11 +5,17 @@
 declare class Thing {
 	foo(): number;
 }
+declare namespace Thing {
+	interface SubThing { }
+}
 export = Thing;
 export as namespace Foo;
 
 // @filename: a.ts
 /// <reference path="foo.d.ts" />
-let y: Foo;
-y.foo();
+import * as ff from './foo';
 
+let y: Foo; // OK in type position
+y.foo();
+let z: Foo.SubThing; // OK in ns position
+let x: any = Foo; // Not OK in value position

